pkg/sql/opt/memo/statistics_builder.go

```diff
@@ -2102,7 +2106,7 @@ func (sb *statisticsBuilder) selectivityFromDistinctCounts(
 		oldDistinct := inputStat.DistinctCount
 
 		if oldDistinct != 0 && newDistinct < oldDistinct {
-			selectivity *= newDistinct / oldDistinct
+			selectivity *= min(newDistinct/oldDistinct, unknownFilterSelectivity)
```

Comment:
This `min` check will probably end up biting us later on, when we have more accurate distinct counts, and yet can never compute a selectivity > 1/3. I see why you have it here, and think it's right way to go to solve the short-term problem. But longer-term, it could be one of many small changes that seem harmless in isolation, but combine together to make the code fragile and filled with special cases.